ioctl/cmd/bc/bcbucketlist.go

```diff
@@ -98,59 +153,55 @@ func getBucketList(arg string) error {
 	return nil
 }
 
-func getBucketListByAddress(addr string) (*iotextypes.VoteBucketList, error) {
-	conn, err := util.ConnectToEndpoint(config.ReadConfig.SecureConnect && !config.Insecure)
-	if err != nil {
-		return nil, output.NewError(output.NetworkError, "failed to connect to endpoint", err)
-	}
-	defer conn.Close()
-	cli := iotexapi.NewAPIServiceClient(conn)
-	method := &iotexapi.ReadStakingDataMethod{
-		Method: iotexapi.ReadStakingDataMethod_BUCKETS_BY_VOTER,
-	}
-	methodData, err := proto.Marshal(method)
-	if err != nil {
-		return nil, output.NewError(output.SerializationError, "failed to marshal read staking data method", err)
-	}
+// getBucketList get bucket list from chain by voter address
+func getBucketListByAddress(addr string, offset, limit uint32) (*iotextypes.VoteBucketList, error) {
 	readStakingdataRequest := &iotexapi.ReadStakingDataRequest{
 		Request: &iotexapi.ReadStakingDataRequest_BucketsByVoter{
 			BucketsByVoter: &iotexapi.ReadStakingDataRequest_VoteBucketsByVoter{
 				VoterAddress: addr,
 				Pagination: &iotexapi.PaginationParam{
-					Offset: uint32(0),
-					Limit:  uint32(1000),
+					Offset: offset,
+					Limit:  limit,
 				},
 			},
 		},
 	}
-	requestData, err := proto.Marshal(readStakingdataRequest)
+	return GetBucketList(iotexapi.ReadStakingDataMethod_BUCKETS_BY_VOTER, readStakingdataRequest)
+}
+
+// getBucketListByCand get bucket list from chain by candidate name
+func getBucketListByCand(arg string, offset, limit uint32) error {
+	bl, err := getBucketListByCandidateName(arg, offset, limit)
```

Comment:
think no need for an extra getBucketListByCandidateName()
directly use the code here

Reply:
so it is similar to above getBucketListByVoter --> getBucketListByAddress